### DISCUSSION TITLE
How to push string array into vector ?

### DISCUSSION THREAD

**Discussion by Ghonghito** [aptos-labs/aptos-developer-discussions#112] [Questions] [answered]
Hello, I'm new to the Move language and I don't understand how to add an array of strings in a vector. So, I made a contract that registers wallet addresses.

My issue is that when a user calls the registerWallets contract with, for example, the argument [a, b], it saves this [a, b] array. However, when the user calls it again with different arguments [c, d], it only saves [c, d] and not [a, b, c, d].

`module WalletRegistrator::walletregister {
    use std::signer;
    use std::string;
    
    struct UserWallets has key {
        walletAddress: vector<string::String>,
    }

    public entry fun registerWallets(caller: &signer, walletAddress: vector<string::String>) acquires UserWallets {
        let caller_address = signer::address_of(caller);
        if (!exists<UserWallets>(caller_address)) {
            move_to(caller, UserWallets {
                walletAddress: walletAddress,
            })
        } else {
          let account_addresses = borrow_global_mut<UserWallets>(caller_address);
          account_addresses.walletAddress = walletAddress
        }
    }

    #[view]
    public fun get_registered_wallets(addr: address): vector<string::String> acquires UserWallets {
        borrow_global<UserWallets>(addr).walletAddress
    }
    
}`

**Comment by gregnazario** [comment, marked as answer]
## How to append a vector onto another vector

So what you're trying to accomplish is like most languages where you want to do the following:

```
vector[1,2,3] + vector[4,5,6] = vector[1,2,3,4,5,6]
```

You can do this with the `vector::append()` See https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/move-stdlib/sources/vector.move#L100-L104

### How do you apply it to your example

```move
struct UserWallets has key {
    walletAddress: vector<string::String>,
}

public entry fun registerWallets(caller: &signer, walletAddress: vector<string::String>) acquires UserWallets {
    let caller_address = signer::address_of(caller);
    if (!exists<UserWallets>(caller_address)) {
        move_to(caller, UserWallets {
            walletAddress: walletAddress,
        })
    } else {
      let account_addresses = borrow_global_mut<UserWallets>(caller_address);
      
      // Simply use append here
      vector::append(&mut account_addresses.walletAddress, walletAddress);
    }
}

#[view]
public fun get_registered_wallets(addr: address): vector<string::String> acquires UserWallets {
    borrow_global<UserWallets>(addr).walletAddress
}
```

**Reply by Ghonghito** [reply]
Thank you! It works very well! <3 I have another question. When a user calls registerWallets, I want to store registered wallets in two separate lists:

1) You already helped me with this <3
2) In a list that acts as a global store, so when is_registered_wallet is called with a wallet address argument (which is a string and not an address), it returns a boolean true or false.
At first, I tried registering wallet addresses like this:

I thought this would register addresses into the global store, but I'm getting a 'Generic Error' in the Petra wallet when trying to register.

```
struct RegisterdWallets has key {
        walletAddresses: vector<string::String>,
}

public entry fun registerWallets(caller: &signer, walletAddress: vector<string::String>) acquires UserWallets, RegisterdWallets {
        let caller_address = signer::address_of(caller);
        if (!exists<UserWallets>(caller_address)) {
            move_to(caller, UserWallets {
                walletAddress: walletAddress,
            })
        } else {
            let account_addresses = borrow_global_mut<UserWallets>(caller_address);
            let registered_addresses = borrow_global_mut<RegisterdWallets>(@WalletRegistrator);
            // Simply use append here
            vector::append(&mut account_addresses.walletAddress, walletAddress);
            vector::append(&mut registered_addresses.walletAddresses, walletAddress);
        }
}
```

and then i have this function:

also getting error when calling this function

```
#[view]
    public fun is_registered_wallet(addr: string::String): bool acquires RegisterdWallets {
        let addressses = &borrow_global<RegisterdWallets>(@WalletRegistrator).walletAddresses;
        vector::contains(addressses, &addr)
}
```
